File: server/src/Models/DMToolkitModels/Items/Definitions/WeaponDefinition.cs
using DMToolkit.API.Models.DMToolkitModels.Entities;
using DMToolkit.API.Models.DMToolkitModels.Items.Bases;
using SharedModels.Enums;

namespace DMToolkit.API.Models.DMToolkitModels.Items.Definitions;

public class WeaponDefinition : ItemDefinitionBase, IItemDefinition
{
    public WeaponCategory WeaponCategory { get; set; } = WeaponCategory.Simple;
    public int NumberOfDice { get; set; } = 0;
    public int NumberOfSides { get; set; } = 0;
    public string DamageType { get; set; } = string.Empty;
    public ICollection<Effect> WeaponProperties { get; set; } = new List<Effect>();
    public string? WeaponMasteryId { get; set; } = null; 
    public Effect? WeaponMastery { get; set; } = null;

    public WeaponDefinition()
    {
        ItemType = "Weapon";
    }
}

File: web-client/src/lib/types/dm-tool-types/items/definitions/weaponDefinition.ts
import type { Effect } from "../../entities/effect";
import type { ItemDefinitionBase, ItemDefinitionBaseDto } from "../bases/itemDefinitionBase";

export interface WeaponDefinition extends ItemDefinitionBase {
    numberOfDice: number;
    numberOfSides: number;
    damageType: string;
    isMartial: boolean;
    weaponProperties: Effect[];
    weaponMastery: Effect | null;
}

export interface WeaponDefinitionDto extends ItemDefinitionBaseDto {
    numberOfDice: number;
    numberOfSides: number;
    damageType: string;
    isMartial: boolean;
    weaponPropertyIds: string[];
    weaponMasteryId: string;
}